mlir-tensorrt/executor/lib/Runtime/API/API.cpp

```diff
@@ -691,6 +729,17 @@ StatusOr<std::unique_ptr<MemRefValue>> MemRefValue::create(
     return getInvalidArgStatus("a specific device must be provided for MemRefs "
                                "that are device-visible");
 
+  // Check if given strides match canonical stride
+  if (!strides.empty() && !shape.empty()) {
+      std::vector<int64_t> canonicalStride = getCanonicalStride(shape);
+      if (!areStridesEquivalent(shape, strides, canonicalStride)) {      
```

Comment:
do not merge, I suspect there's some miscommunication here, let's discuss offline